File: python/509.py
#509. The Fibonacci numbers, commonly denoted F(n) form a sequence, called the Fibonacci sequence, such that each number is the sum of the two preceding ones, starting from 0 and 1. That is,
from typing import List
def fib(n: int) -> int:
    lst = [0]
    while(len(lst) != n+1):
        if(len(lst) == 1): lst.append(1)
        else: lst.append(lst[len(lst)-2] + lst[len(lst)-1])
    return lst[n]

n = int(input())
print(fib(n))

File: golang/509.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(fib(2))
	fmt.Println(fib(3))
	fmt.Println(fib(6))
}

// DESC: find fibbonacci sequence F(0) = 0, F(1) = 1
// F(n) = F(n+1) + F(n+2)
func fib(n int) int {
	if n == 0 {
		return 0
	}
	if n == 1 {
		return 1
	}
	return fib(n-1) + fib(n-2)
}
